lib/OpenQA/CacheService/Model/Cache.pm

```diff
@@ -48,166 +48,132 @@ sub from_worker {
-        else {
-            $log->debug('CACHE: Abnormal situation, code 304. Retrying download');
-            $asset = 520;
+        $log->info("Content of $asset has not changed, updating last use");
+        unless ($self->_update_asset_last_use($asset)) {
+            $log->warn('Abnormal situation, status 304');
+            $ret = 520;
         }
     }
-    elsif ($tx->res->is_server_error) {
-        $log->debug("CACHE: Could not download the asset, triggering a retry for $code.");
-        $log->debug("CACHE: Abnormal situation, server error. Retrying download");
-        $asset = $code;
+
+    elsif ($res->is_server_error) {
+        $log->warn("Abnormal situation, server error $code");
+        $ret = $code;
     }
-    elsif ($tx->res->is_success) {
+
+    elsif ($res->is_success) {
+        my $headers = $tx->res->headers;
         $etag = $headers->etag;
-        unlink($asset);
+        unlink $asset;
         $self->_cache_sync;
-        my $size = $tx->res->content->asset->move_to($asset)->size;
+        my $size = $res->content->asset->move_to($asset)->size;
         if ($size == $headers->content_length) {
             $self->_check_limits($size);
 
             # This needs to go in to the database at any cost - we have the lock and we succeeded in download
             # We can't just throw it away if database locks.
             my $att = 0;
             my $ok;
-            ++$att and sleep 1 and $log->debug("CACHE: Error updating Cache: attempting again: $att")
+            ++$att and sleep 1 and $log->warn("Updating cache failed (attempt $att)")
               until ($ok = $self->_update_asset($asset, $etag, $size)) || $att > 5;
-            unless ($ok) {
-                $log->error('CACHE: FAIL Could not update DB - purging asset');
+
+            if ($ok) { $log->info("Download of $asset successful, new cache size is $self->{cache_real_size}") }
+            else {
+                $log->error("Purging $asset because updating the cache failed, this should never happen");
                 $self->purge_asset($asset);
-                $asset = undef;
             }
-            $log->debug("CACHE: Asset download successful to $asset, Cache size is: $self->{cache_real_size}") if $ok;
         }
         else {
-            $log->debug(
-                "CACHE: Size of $asset differs, Expected: " . $headers->content_length . " / Downloaded: " . "$size");
-            $asset = 598;    # 598 (Informal convention) Network read timeout error
+            $log->warn("Size of $asset differs, expected " . $headers->content_length . " but downloaded $size");
+            $ret = 598;    # 598 (Informal convention) Network read timeout error
         }
     }
     else {
-        my $message = $tx->res->error->{message};
-        $log->debug("CACHE: Download of $asset failed with: $code - $message");
+        my $message = $res->error->{message};
+        $log->warn("Purging $asset because the download failed: $code - $message");
```

Comment:
So making this a warn up from debug means to me you consider this a situation we should react on?

Reply:
It's not an error but someone should probably take a look, yes, i've seen this mostly raise 404 for missing ISOs so far.

Reply:
I see. Let say, that's a point for improvement later on to separate better "user errors" from something that either the openQA dev or an instance admin would need to be aware of. E.g. a missing ISO I would consider as "user error" as it is very easy to erroneously schedule a job for a missing asset.